unit_tests/src/test_connection_pool.cpp

```diff
@@ -261,4 +262,79 @@ TEST_CASE("connection pool")
             REQUIRE(e.code() == SYS_SOCK_CONNECT_ERR);
         }
     }
+
+    SECTION("connection_pool_options")
+    {
+        irods::connection_pool_options cp_opts{};
+
+        SECTION("refresh connection after N retrievals")
+        {
+            cp_opts.refresh_connections_after_n_retrievals = 3;
+
+            irods::experimental::fully_qualified_username user{env.rodsUserName, env.rodsZone};
+            irods::connection_pool conn_pool{1, env.rodsHost, env.rodsPort, user, cp_refresh_time, cp_opts};
+
+            RcComm* conn_ptr{};
+
+            {
+                auto conn = conn_pool.get_connection();
+                CHECK(conn);
+
+                // Capture the memory address of the underlying RcComm.
+                conn_ptr = static_cast<RcComm*>(conn);
+                CHECK(conn_ptr);
+            }
+
+            {
+                auto conn = conn_pool.get_connection();
+                CHECK(conn);
+
+                // Show that the memory address has not yet changed.
+                CHECK(static_cast<RcComm*>(conn) == conn_ptr);
+            }
+
+            {
+                auto conn = conn_pool.get_connection();
+                CHECK(conn);
+
+                // Show that the memory address has not yet changed.
+                CHECK(static_cast<RcComm*>(conn) == conn_ptr);
+            }
+
+            auto conn = conn_pool.get_connection();
+            CHECK(conn);
+
+            // Show that the memory address has changed now that the retrieval count has been exceeded.
+            CHECK(static_cast<RcComm*>(conn) != conn_ptr);
+        }
+
+        SECTION("refresh connection after resource modification")
+        {
+            cp_opts.refresh_connections_after_resource_updates = true;
+
+            irods::experimental::fully_qualified_username user{env.rodsUserName, env.rodsZone};
+            irods::connection_pool conn_pool{1, env.rodsHost, env.rodsPort, user, cp_refresh_time, cp_opts};
+
+            namespace adm = irods::experimental::administration;
+
+            RcComm* conn_ptr{};
+
+            {
+                auto conn = conn_pool.get_connection();
+                REQUIRE(conn);
+
+                conn_ptr = static_cast<RcComm*>(conn);
+                REQUIRE(conn_ptr);
+
+                // This will cause all connections in the pool to be replaced on their next usage.
+                REQUIRE_NOTHROW(adm::client::modify_resource(
+                    conn, "demoResc", adm::resource_comments_property{"testing connection pool refresh"}));
```

Comment:
Is it okay to assume `demoResc` here?

The tests are mostly designed to run in a known pre-configured environment. I think I'm okay with this.

Reply:
lots of tests assume this.   i think it's okay.

Reply:
Great. Leaving as is then.